tests/testthat/test-as-data-frame.R

```diff
@@ -76,6 +76,63 @@ with_mock_crunch({
         expect_identical(evalq(gender, as.data.frame(ds)),
             as.vector(ds$gender))
     })
+    csv_df <- read.csv("dataset-fixtures/test_ds.csv", stringsAsFactors = FALSE, colClasses = "character")
+    test_that("csvToDataFrame produces the correct data frame", {
+        expected <- readRDS("dataset-fixtures/test_ds.rds")
+        cdf <- as.data.frame(ds[, c("birthyr", "gender", "location", "mymrset", "textVar", "starttime")])
+        #test local CDF variables
+        cdf$newvar <- expected$newvar <- 1:25
+        expect_identical(csvToDataFrame(csv_df, cdf), expected)
+    })
+
+    test_that("csvToDataFrame handles hidden variables", {
+        new_ds <- loadDataset("test ds")[, c("birthyr", "gender", "location", "mymrset", "textVar", "starttime")]
+        new_ds$birthyr@tuple[["discarded"]] <- TRUE
+        new_ds_df <- as.data.frame(new_ds)
+        expect_silent(
+            expect_equal(names(csvToDataFrame(csv_df, new_ds_df)),
+                c("gender", "location", "subvar2", "subvar1", "subvar3", "textVar",
+                    "starttime"))
+        )
+        # now we want the hidden vars to be included
+        new_ds_df <- as.data.frame(new_ds, include.hidden = TRUE)
+        expect_warning(
+            expect_equal(names(csvToDataFrame(csv_df, new_ds_df)),
+                c("birthyr", "gender", "location", "subvar2", "subvar1", "subvar3", "textVar",
+                    "starttime")),
+            paste0("Variable birthyr is hidden"))
+    })
```

Comment:
Nice refactoring for test coverage of the main content here!

You could also add a test to catch the first 4 lines of `as.data.frame.CrunchDataFrame` by testing that the post for the csv download is as expected (cf https://github.com/Crunch-io/rcrunch/blob/master/tests/testthat/test-export-dataset.R#L9-L11). Then the only things that are un-(unit)-tested are: reading the csv and deleting it when you're done.